kanidmd/idm/src/idm/credupdatesession.rs

```diff
@@ -391,70 +406,129 @@ impl<'a> IdmServerProxyWriteTransaction<'a> {
         token: CredentialUpdateIntentToken,
         ct: Duration,
     ) -> Result<CredentialUpdateSessionToken, OperationError> {
-        let token: CredentialUpdateIntentTokenInner = self
-            .token_enc_key
-            .decrypt(&token.token_enc)
-            .map_err(|e| {
-                admin_error!(?e, "Failed to decrypt intent request");
-                OperationError::SessionExpired
-            })
-            .and_then(|data| {
-                serde_json::from_slice(&data).map_err(|e| {
-                    admin_error!(err = ?e, "Failed to deserialise intent request");
-                    OperationError::SerdeJsonError
-                })
-            })?;
-
-        // Check the TTL
-        if ct >= token.max_ttl {
-            trace!(?ct, ?token.max_ttl);
-            security_info!(%token.sessionid, "session expired");
-            return Err(OperationError::SessionExpired);
-        }
+        let CredentialUpdateIntentToken { intent_id } = token;
+
+        /*
+            let entry = self.qs_write.internal_search_uuid(&token.target)?;
+        */
+        // ⚠️  due to a low, but possible risk of intent_id collision, if there are multiple
+        // entries, we will reject the intent.
+        // DO we need to force both to "Consumed" in this step?
```

Comment:
I'm not sure I'd use `Consumed` as the result, this feels like it needs its own state so it's really clear what happened - maybe a `ForcedExpired` so that if say ... a collision happened, or an admin killed the intent token at some point, it could be logged + terminated.

Reply:
Yeah, that was my thinking was a seperate collision state, but for now, it just forces to a safe failure condition at least.